### DISCUSSION TITLE
Not connecting to WIFI.

### DISCUSSION THREAD

**Discussion by st2000** [tjko/brickpico#19] [Q&A] [answered]
Just put together 1st of 3 of these. I set the SSID & the PASSWORD.  But sys:wifi:stat? only returns: "1,0.0.0.0,255.255.255.0,192.168.0.1". It has been sitting here on the workbench or over an hour. Don't think the SSID or PASSWORD are incorrect. I've tried twice. And many other "home programmed" devices (i.e. Tasmota & ESPNow like devices) are working off the same WIFI SSID. Some were / are in the same location (WRT the distance to the WIFI/Router). I believe the WIFI/Router is running WPA.

**Comment by st2000** [comment]
Looks like if I change from Asynchronous to Synchronous connect mode ("SYS:WIFI:MODE 1") I see that I get a bit farther in the boot up messages (entries WRT connecting to a WIRI network). However, it appears to always fail w/error "WiFi connect failed: -1". After that, it appears the PICO is stuck in or just after posting "Initialize GPIO...". Yes, my WIFI router is old. But I've never had such a problem before. And I run a bunch of embedded open source and proprietary WIFI IOT devices.

**Reply by tjko** [reply]
Does your router only support WPA (and not WPA2/WPA3)?

Looks like current firmware uses WPA2, so it could be that connection to Access Point that only support WPA will indeed fail.

I was going to update brickpico firmware to support new Pico-SDK v2.1.0, I'll see if I can also add configuration option to specify WiFi authorization type (Open, WPA, WPA2, WPA/WPA2, etc...)

Btw, are you able to compile the firmware from sources?  It would be interesting to know if  changing ```CYW43_AUTH_WPA2_AES_PSK``` to ```CYW43_AUTH_WPA2_MIXED_PSK``` in  _src/network.c_ (there is two lines 166 & 176 to update) might help, as it should allow both WPA2 and WPA being used...

**Comment by tjko** [comment, marked as answer]
@st2000, there is new "beta" firmware available here: https://github.com/tjko/brickpico/actions/runs/12476677378
(there should link to firmware bundle zip towards the bottom of the page)

It adds new command SYS:WIFI:AUTH that can be used to configure WiFi Authentication Mode.
It allows configuring any of the supported modes on Pico W: https://github.com/tjko/brickpico/blob/main/commands.md#systemwifiauthmode


If your AP only support "WPA", then setting authentication mode to WPA might help:
```
SYS:WIFI:AUTH WPA
CONF:SAVE
*RST
```

**Comment by st2000** [comment]
I am having a heck of a time getting the "blink" example to compile let alone brickpico.

I thought I knew how to use VSCode. But this raspberry-pi-pico integration and all the prerequisites, not to mention PICO HW oddities on the WIFI version (messed up GPIO so the LED is actually connected to the WIFI module??)... or is all that just old news that I'm getting caught in. Regardless, I can't try your suggested alteration to your code as I just can not connect the VSCode/PICO dots.

As for your new binaries, they worked! I'm appreciative and a little taken aback. I'm looking at my WIFI router's configuration page and it says "WPA-2". But I did have to set the BrickPICO to WPA. Maybe this is way this Arris/AT&T/Motorola (that's just too many cooks) ADSL-Modem/Router has earned a great deal of disrespect down through the years.

So...

Yes, the new binaries work. Thanks!

If I ever get this VSCode/PICO environment to work I can take a look at some type of flashing duty cycle / interval thing (i.e. new airplanes). Maybe even an attack / decay thing (i.e. old airplanes).

**Reply by st2000** [reply]
... finally got a local PICO tool chain running on a Ubuntu desktop. Turns out I missed the bit requiring me to grab the pico-sdk git submodules.
